src/whisper_at_train/run_combined_training.sh

```
@@ -7,11 +7,28 @@
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+PROJECT_ROOT="$(cd "${SCRIPT_DIR}/../.." && pwd)"
+
+# Environment setup - check if running in Poetry environment
+if command -v poetry &> /dev/null && poetry env info --path &> /dev/null; then
+    echo "Using Poetry environment"
+    poetry shell
+else
+    # Fallback to virtual environment activation
+    VENV_PATH="${VENV_PATH:-/home/taemyung_heo/.cache/pypoetry/virtualenvs/whisper-at-z6hdRBdT-py3.10/bin/activate}"
+    if [[ -f "$VENV_PATH" ]]; then
+        source "$VENV_PATH"
```

Shellcheck (0.10.0)

[warning] ShellCheck can't follow non-constant source. Use a directive to specify location.

(SC1090)